### PR TITLE
Java9update

### PR DESCRIPTION
**Pull request by anddann**
Merge the new LambdaMetaFactory Feature as implemented by @mbenz89 back into Soot's Java9 Branch
https://soot-build.cs.uni-paderborn.de/jenkins/job/soot/job/soot-java9update/

### CODE REVIEW

**Comment by anddann** [comment]
Before merging this into the Java9 branch: 1. merge PR #1109 into develop, 2. merge develop into this PR.

**Comment by anddann** [comment]
@mbenz89 Would you mind taking a look at this PR?

File: doc/soot_options.htm [hunk 1]
```diff
@@ -1083,7 +1085,7 @@ <h2><a name="phase_1">Jimple Body Creation (jb)
             </a></h2>
       <p>
          Jimple Body Creation creates a JimpleBody for each input
-         method, using either coffi, to read .class files, or the
+         method, using either asm, to read .class files, or the
```
Comment (mbenz89) [inline doc/soot_options.htm]: great!

File: pom.xml [hunk 2]
```diff
@@ -18,21 +18,54 @@
             <distribution>repo</distribution>
         </license>
     </licenses>
+    <developers>
+        <developer>
+            <name>Manuel Benz</name>
```
Comment (mbenz89) [inline pom.xml]: Nice change. I like that one!

File: doc/soot_options.htm (hunk 1 above)
Comment (anddann) [reply doc/soot_options.htm]: Super great!

File: pom.xml (hunk 2 above)
Comment (anddann) [reply pom.xml]: Yes, this and the change below are my two favorite changes! :1st_place_medal:

**Comment by anddann** [comment]
[x] Merge new asm version into java9update:
[x] test it on jenkins (we need a Jenkinsfile or JDK 11)
[] merge the branch java9update #1107 into the java9 branch
[] Merge Java9 into develop (someday) 1st_place_medal

**Pull request merged**